protocol/parser/parser.go

```diff
@@ -261,28 +262,32 @@ func (p *ParsedInput) GetBlockHashes() ([]string, error) {
 	return p.parsedHashes, nil
 }
 
-func ParseWithGenericParsers(rpcInput RPCInput, genericParsers []spectypes.GenericParser) (*ParsedInput, error) {
+func getMapForParse(rpcInput RPCInput) map[string]interface{} {
+	return map[string]interface{}{"params": rpcInput.GetParams(), "result": rpcInput.GetResult()}
+}
+
+func ParseWithGenericParsers(rpcInput RPCInput, genericParsers []spectypes.GenericParser, parserType string) (*ParsedInput, error) {
 	if len(genericParsers) == 0 {
 		return nil, fmt.Errorf("no generic parsers to use")
 	}
 
-	params := rpcInput.GetParams()
-	if params == nil {
-		return nil, utils.LavaFormatTrace("rpcInput params are nil", utils.LogAttr("rpcInput", rpcInput))
-	}
+	parsingMap := getMapForParse(rpcInput)
 
 	// We try to parse the params with all the generic parsers, the first one that succeeds, its value is returned
 	for _, genericParser := range genericParsers {
-		retval, err := parseGeneric(params, genericParser)
+		// skip generic parsers which do not fit current parser type
+		if !strings.HasPrefix(genericParser.ParsePath, parserType) {
```

Comment:
check the parser enum to be default value, or block latest, when parser type is BLOCK_PARSING